core/src/avm1/activation.rs

```diff
@@ -899,24 +871,21 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
 
     fn action_define_function(
         &mut self,
-        name: &'_ SwfStr,
-        params: &[&'_ SwfStr],
-        actions: SwfSlice,
+        action: DefineFunction2,
+        parent_data: &SwfSlice,
     ) -> Result<FrameControl<'gc>, Error<'gc>> {
-        let name = name.to_str_lossy(self.encoding());
-        let name = name.as_ref();
         let swf_version = self.swf_version();
+        let func_data = parent_data.to_unbounded_subslice(action.actions).unwrap();
         let scope = Scope::new_closure_scope(self.scope_cell(), self.context.gc_context);
         let constant_pool = self.constant_pool();
-        let func = Avm1Function::from_df1(
+        let func = Avm1Function::from_swf_function(
             self.context.gc_context,
             swf_version,
-            actions,
-            name,
-            params,
+            func_data,
+            action,
             scope,
             constant_pool,
-            self.target_clip_or_root()?,
+            self.base_clip(),
```

Comment:
Note to self, check this because I think this was a specific difference between DefineFunction (SWF5) and DefineFunction2 (SWF6+).

Reply:
Looks like this is dependent on SWF version, not DefineFunction tag version. Should use base clip in SWFv6, target clip otherwise. Need to add an `swf_version` check here.

Reply:
Ok, just checked and I had already taken care of in [`Executable::exec`](https://github.com/ruffle-rs/ruffle/blob/53589407740a724a97204efd3d504d108dff8f2e/core/src/avm1/function.rs#L318-L323). So `action_define_function` was actually incorrect in SWF6 before, and this PR fixes the following example:

```
	tellTarget(clip) {
		f = function() {
			trace(_target);
		}
	}
	
	clip.f(); // Should be "/", traces "/clip" before this PR.
```
There are some other edge cases here, I'll make a test case in a future PR.